pkg/storage/bucket/azure/bucket_client.go

```diff
@@ -7,28 +7,33 @@ package azure
 
 import (
 	"github.com/go-kit/log"
+	"github.com/pkg/errors"
 	"github.com/thanos-io/objstore"
 	"github.com/thanos-io/objstore/providers/azure"
 	yaml "gopkg.in/yaml.v3"
 )
 
-func NewBucketClient(cfg Config, name string, logger log.Logger) (objstore.Bucket, error) {
+func NewBucketClient(cfg Config, name string, logger log.Logger, factory func(log.Logger, []byte, string) (*azure.Bucket, error)) (objstore.Bucket, error) {
```

Comment:
A post-merge comment. There's no need to take the `factory` in the exported function. What we do in other places is:
1. Create `newBucketClient()` (non exported) taking the `factory`. Test this.
2. Keep `NewBucketClient()` (exported) with **no** `factory`. It just calls `newBucketClient()` passing the `factory` too.